dash/include/dash/atomic/GlobAtomicRef.h

```diff
@@ -320,39 +320,65 @@ class GlobRef<dash::Atomic<T>>
    */
   T fetch_sub (
     /// Value to be subtracted from global atomic variable.
-    const T & value) const
+    const T & value)
   {
     return fetch_op(dash::plus<T>(), -value);
   }
 
+  /**
+   * DASH specific variant which is faster than \c fetch_multiply
+   * but does not return value
+   */
+  void multiply(const T & value)
+  {
+    op(dash::multiply<T>(), value);
+  }
+
+  /**
+   * Atomic fetch-and-multiply operation on the referenced shared value.
+   *
+   * \return  The value of the referenced shared variable before the
+   *          operation.
+   */
+  T fetch_multiply(
+    /// Value to be added to global atomic variable.
+    const T & value)
+  {
+    return fetch_op(dash::multiply<T>(), value);
+  }
+
   /// prefix atomically increment value by one
-  T operator++ () const {
-    return fetch_add(1) + 1;
+  self_t& operator++ () {
```

Comment:
This is a highly problematic change, as the operator is not atomic any more. I thought about this for a while as this also affects the coarray correctness. There are use-cases where returning a reference is necessary
 (`++(++a))`) however even the STL does not support this for [std::atomic](http://en.cppreference.com/w/cpp/atomic/atomic/operator_arith).

If you like we can have a discussion on that, but I strongly recommend to revert this to the current semantics.

Reply:
Thanks a lot for the clarification. I restored the original behavior and documented it for future wanderers wondering about it :)